### PR TITLE
Create basic models

### PR DESCRIPTION
**Pull request by assafgelber**
We just need to update the flows to work with these models instead of the ad-hoc ones we made.

### CODE REVIEW

File: app/src/main/java/com/iic/shopingo/dal/models/Contact.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package com.iic.shopingo.dal.models;
+
+/**
+ * Created by assafgelber on 3/8/15.
+ */
+public class Contact {
+  private String firstName;
+  private String lasttName;
```
Comment (AsafGartner) [inline app/src/main/java/com/iic/shopingo/dal/models/Contact.java]: double T

File: app/src/main/java/com/iic/shopingo/dal/models/BaseRequest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    DECLINED,
+    CANCELED
+  }
+
+  private Contact contact;
+  private ShoppingList shoppingList;
+  private RequestStatus status;
+
+  public BaseRequest(Contact contact, ShoppingList shoppingList, RequestStatus status) {
+    this.contact = contact;
```
Comment (AsafGartner) [inline app/src/main/java/com/iic/shopingo/dal/models/BaseRequest.java]: I think contact shouldn't be in base.
You should have `shopper` in `OutgoingRequest` and `requester` in `IncomingRequest`.

Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/dal/models/BaseRequest.java]: :+1:

File: app/src/main/java/com/iic/shopingo/dal/models/Contact.java (hunk 1 above)
Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/dal/models/Contact.java]: Whoops, nice catch.

**Comment by agelber** [comment]
Oh wait, they all need to be `Parcelable`, no?

**Comment by ifeins** [comment]
Looks good

**Pull request merged**